src/commands/fuelup.rs

```diff
@@ -0,0 +1,81 @@
+use anyhow::{bail, Result};
+use clap::Parser;
+use tracing::info;
+
+use crate::{
+    constants::{FUELUP_RELEASE_DOWNLOAD_URL, GITHUB_API_REPOS_BASE_URL, RELEASES_LATEST},
+    download::{download_file_and_unpack, fuelup_bin_dir, get_latest_tag, unpack_extracted_bins},
+};
+
+#[derive(Debug, Parser)]
+pub enum FuelupSubcommand {
+    /// Updates fuelup
+    Update,
+}
+
+#[derive(Debug, Parser)]
+struct UpdateCommand {}
+
+fn fuelup_bin_tarball_name(version: &str) -> Result<String> {
+    let architecture = match std::env::consts::ARCH {
+        "aarch64" => "aarch64",
+        "x86_64" => "x86_64",
+        unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
+    };
+
+    let vendor = match std::env::consts::OS {
+        "macos" => "apple",
+        _ => "unknown",
+    };
+
+    let os = match std::env::consts::OS {
+        "macos" => "darwin",
+        "linux" => "linux-gnu",
+        unsupported_os => bail!("Unsupported os: {}", unsupported_os),
+    };
+
+    Ok(format!(
+        "fuelup-{}-{}-{}-{}.tar.gz",
+        // strip the 'v' from the version string to match the file name of the releases
+        &version[1..version.len()],
+        architecture,
+        vendor,
+        os
+    ))
+}
+
+pub fn self_update() -> Result<()> {
+    let proc = std::process::Command::new("fuelup")
+        .arg("--version")
+        .output()
+        .expect("Could not run fuelup.");
```

Comment:
Is there any way to access the version of the current `fuelup` clap app instance that's running without re-invoking it using `Command`?

Reply:
Answering my own question as I remembered we already do this in forc in order to work out the sway git tag there. Check out this macro:

https://docs.rs/clap/latest/clap/macro.crate_version.html

Reply:
this is super helpful, cut down quite a lot of code from this